File: src/utils/academicYear.js
export const ACADEMIC_YEAR_START_MONTH = 9;

export const getDefaultAcademicYear = (date = new Date()) => {
  const year = date.getFullYear();
  const month = date.getMonth() + 1;
  return month >= ACADEMIC_YEAR_START_MONTH ? year : year - 1;
};

export const buildAcademicYearOptions = (
  baseYear = getDefaultAcademicYear(),
  before = 2,
  after = 2
) => {
  const years = [];
  for (let i = -before; i <= after; i += 1) {
    years.push(baseYear + i);
  }
  return years.sort((a, b) => b - a);
};

File: app/utils/academic_year.py
from datetime import datetime


ACADEMIC_YEAR_START_MONTH = 9


def get_default_academic_year(now=None):
    current = now or datetime.now()
    return (
        current.year
        if current.month >= ACADEMIC_YEAR_START_MONTH
        else current.year - 1
    )
